src/ert/shared/models/ensemble_smoother.py

```diff
@@ -24,6 +34,175 @@ def setAnalysisModule(self, module_name: str) -> None:
+        loop = asyncio.get_running_loop()
+        threadpool = concurrent.futures.ThreadPoolExecutor()
+
+        prior_context = await loop.run_in_executor(threadpool, self.create_context)
+
+        experiment_logger.debug("starting ensemble smoother experiment")
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        self._checkMinimumActiveRealizations(prior_context)
+
+        await loop.run_in_executor(
+            threadpool,
+            self.ert().createRunPath,
+            prior_context,
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_SIMULATION, prior_context.iteration, loop, threadpool
+        )
+
+        # Post ensemble, parameters, observations to new storage
+        # (nb: this calls self.setPhaseName())
+        ensemble_id = await loop.run_in_executor(threadpool, self._post_ensemble_data)
+
+        experiment_logger.debug("evaluating")
+        await self._evaluate(prior_context, evaluator_server_config)
+
+        # Push simulation results to storage
+        await loop.run_in_executor(threadpool, self._post_ensemble_results, ensemble_id)
+
+        num_successful_realizations = await self.successful_realizations(
+            prior_context.iteration
+        )
+
+        self.checkHaveSufficientRealizations(num_successful_realizations)
+
+        await self._run_hook(
+            HookRuntime.POST_SIMULATION, prior_context.iteration, loop, threadpool
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_FIRST_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        await self._run_hook(
+            HookRuntime.PRE_UPDATE, prior_context.iteration, loop, threadpool
+        )
+
+        await self.dispatch(
+            CloudEvent(
+                {
+                    "type": identifiers.EVTYPE_EXPERIMENT_ANALYSIS_STARTED,
+                    "source": f"/ert/experiment/{self.id_}",
+                    "id": str(uuid.uuid1()),
+                }
+            ),
+            prior_context.iteration,
+        )
+
+        experiment_logger.debug("running update...")
+        try:
+            await loop.run_in_executor(
+                threadpool, self.facade.smoother_update, prior_context
+            )
+        except ErtAnalysisError as e:
+            experiment_logger.exception("analysis failed")
+            await self.dispatch(
+                CloudEvent(
+                    {
+                        "type": identifiers.EVTYPE_EXPERIMENT_FAILED,
+                        "source": f"/ert/experiment/{self.id_}",
+                        "id": str(uuid.uuid1()),
+                    },
+                    {
+                        "error": str(e),
```

Comment:
Wondering where this `error` field is actually used?

Reply:
Not sure, but I guess it does make sense to pass the error information on. The same data argument is used for hook names as far as I can see, maybe not picked up (yet!) either(?)

Reply:
I see. I wonder mainly whether this kind of general "data" field should be present for Protobuf objects (which of-course it can't, but we can supplement some `error` fields at least) ...
Anyway, for this I understand you would like to keep regardless it's being consumed or not?